react-accordion/my-app/src/TopicCard.tsx

```diff
@@ -1,20 +1,17 @@
 type Props = {
-  topics: { id: number; title: string; content: string }[];
-  handleClick: (selectedTopic: number) => void;
-  selectedTopic: number;
+  topic: { id: number; title: string; content: string };
+  onClick: (clickedTopic: number) => void;
+  selectedTopic: number | undefined;
 };
 
-export function TopicCard({ topics, handleClick, selectedTopic }: Props) {
-  return (
+export function TopicCard({ topic, onClick, selectedTopic }: Props) {
+  const topicCard = (
     <div>
-      {topics.map((topic) => (
-        <div key={topic.id}>
-          <h1 className="topic-header" onClick={() => handleClick(topic.id)}>
-            {topic.title}
-          </h1>
-          {selectedTopic === topic.id ? <p>{topic.content}</p> : <></>}
-        </div>
-      ))}
+      <h1 className="topic-header" onClick={() => onClick(topic.id)}>
+        {topic.title}
+      </h1>
+      {selectedTopic === topic.id && <p>{topic.content}</p>}
     </div>
   );
+  return <div>{topicCard}</div>;
```

Comment:
I would put all the topic JSX inline here instead of using a variable. I will make the HTML structure easier to follow.